tests/test_pytest_writer.py

```diff
@@ -0,0 +1,121 @@
+"""Tests for behave_writer module."""
+import pathlib
+from types import SimpleNamespace
+
+import pytest
+
+from sphinx_gherkindoc import writer
+import rst_output_pytest as rst_output
+
+
+def _reformat_keywords(rst_lines):
+    new_rst_lines = rst_lines[:]
+    for (no_description_keyword, marker) in (("Background", "-"), ("Examples", "~")):
+        line_to_match = (
+            f":gherkin-{no_description_keyword.lower()}-keyword:"
+            f"`{no_description_keyword}:`"
+        )
+        for i, line in enumerate(new_rst_lines):
+            if line.startswith(line_to_match):
+                new_rst_lines[i] = f"{line_to_match}\n"
+            if new_rst_lines[i - 1] == f"{line_to_match}\n":
+                marker_line = marker * len(line_to_match)
+                new_rst_lines[i] = f"{marker_line}\n\n"
+    return new_rst_lines
+
+
+@pytest.fixture()
+def pytest_rst_output():
+    base_rst_output = SimpleNamespace()
+    for attribute in dir(rst_output):
+        if not attribute.startswith("_"):
+            setattr(
+                base_rst_output,
+                attribute,
+                _reformat_keywords(getattr(rst_output, attribute)),
```

Comment:
Can you explain a bit what's going on here? I was expecting simpler tests just comparing output to expected. What is this reformatting?

Reply:
Thanks for the comments. That makes sense. I think I'd prefer to just have a separate copy of the behave output to compare against, so we can see what the RST will look like if we want, but this works alright too.